### PR TITLE
fix: rainbowkit buttons not working

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
replace focus trap with a custom one.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -19807,7 +19806,7 @@ __metadata:
   languageName: node
   linkType: hard
 
-"focus-trap-react@npm:9.0.2, focus-trap-react@npm:^9.0.2":
+"focus-trap-react@npm:9.0.2":
```
Comment (axeldelafosse) [inline yarn.lock:19809]: I think it's still included because the `modal` package wasn't updated via Bit

**Review by axeldelafosse — APPROVED** [review]
LFG ✨

File: packages/design-system/modal/modal.container.web.tsx [hunk 2]
```diff
@@ -78,4 +77,67 @@ const styles = StyleSheet.create({
...
+      } else {
+        contentRef.current.focus();
+      }
+    }
+  }, [contentRef]);
+
+  return (
+    <>
+      <div
+        onFocus={useCallback(
```
Comment (catalinmiron) [inline packages/design-system/modal/modal.container.web.tsx]: can we move this outside the render?

File: packages/design-system/modal/modal.container.web.tsx [hunk 3]
```diff
@@ -78,4 +77,67 @@ const styles = StyleSheet.create({
...
+        tabIndex={0}
+      />
+      <div
+        ref={contentRef}
+        style={{ outline: "none" }}
+        tabIndex={-1}
+        {...props}
+      />
+      <div
+        onFocus={useCallback(
```
Comment (catalinmiron) [inline packages/design-system/modal/modal.container.web.tsx]: same

**Review by catalinmiron — APPROVED** [review]
nit: please move `useCallback` outside the render

File: packages/design-system/modal/modal.container.web.tsx (hunk 2 above)
Comment (intergalacticspacehighway) [reply packages/design-system/modal/modal.container.web.tsx]: done!

**Pull request merged**